chia/rpc/wallet_rpc_api.py

```diff
@@ -842,14 +843,20 @@ async def create_offer_for_ids(self, request):
             )
         if success:
             return {
-                "offer": trade_record.offer.hex(),
+                "offer": Offer.from_bytes(trade_record.offer).compress().hex(),
                 "trade_record": trade_record.to_json_dict_convenience(),
             }
         raise ValueError(error)
 
     async def get_offer_summary(self, request):
         assert self.service.wallet_state_manager is not None
         offer_hex: str = request["offer"]
```

Comment:
How about we convert to bytes here and avoid the back and forth below?

Reply:
We don't really have precedent for sending non-hex over the RPCs, so I'm going to keep it like this for now.